### PR TITLE
WIP Reconnecting http2 client PR followup

### PR DESCRIPTION
**Pull request by johanandren**
On top of #3685 (would perhaps be better to push directly to that PR but separate for now)

### CODE REVIEW

**Comment by johanandren** [comment]
Failed because of some race condition tearing down I think:

```
 [WARN] [01/08/2021 11:26:11.650] [Http2PersistentClientSpec-akka.actor.default-dispatcher-6] [akka://Http2PersistentClientSpec/system/IO-TCP/selectors/$a/6] unhandled message from Actor[akka://Http2PersistentClientSpec/deadLetters]: ChannelReadable
```

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Could be the same as reported here perhaps? https://github.com/akka/akka/issues/29330

**Review by ignasi35 — APPROVED** [review]

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2PersistentClientSpec.scala [hunk 1]
```diff
@@ -253,11 +258,14 @@ class Http2PersistentClientSpec extends AkkaSpecWithMaterializer(
...
 
-      lazy val clientFlow =
-        Http().connectionTo("akka.example.org")
+      lazy val clientFlow = {
+        val builder = Http().connectionTo("akka.example.org")
           .withCustomHttpsConnectionContext(ExampleHttpContexts.exampleClientContext)
           .withClientConnectionSettings(clientSettings.withTransport(transport))
-          .managedPersistentHttp2()
+
+        if (tls) builder.managedPersistentHttp2()
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2PersistentClientSpec.scala:266]: :+1:

**Comment by jrudolph** [comment]
Thanks @johanandren for getting it over the line.

**Pull request merged**